File: src/main/java/domain/Conference.java
package domain;

import javax.persistence.*;

@Entity
@Table(name="Conferences")
public class Conference {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "conference_id")

    private Integer conferenceID;
    @Column(name = "name",nullable = false)
    private String name;

    @Column(name="start_date",nullable = false)
    @Temporal(TemporalType.DATE)
    java.util.Date startDate;

    @Column(name="end_date",nullable = false)
    @Temporal(TemporalType.DATE)
    java.util.Date endDate;

    @Column(name="proposal_deadline",nullable = false)
    @Temporal(TemporalType.DATE)
    java.util.Date proposalDeadline;

    @Column(name="paper_deadline",nullable = false)
    @Temporal(TemporalType.DATE)
    java.util.Date paperDeadline;

}

File: client/CMS/src/app/shared/models/Conference.ts
export class Conference {
  conferenceID: number;
  name: string;
  startDate: Date;
  endDate: Date;
  proposalDeadline: Date;
  paperDeadline: Date;

  constructor() {
    this.conferenceID = 0;
    this.name = '';
    this.startDate = new Date();
    this.endDate = new Date();
    this.proposalDeadline = new Date();
    this.paperDeadline = new Date();
  }

  public setAttributes(conferenceID, name, startDate, endDate, proposalDeadline, paperDeadline) {
    this.conferenceID = conferenceID;
    this.name = name;
    this.startDate = startDate;
    this.endDate = endDate;
    this.proposalDeadline = proposalDeadline;
    this.paperDeadline = paperDeadline;
  }

  public getName() {
    return this.name;
  }
}

export class Conferences{
  conferenceDtos: Array<Conference>;
}
